backend/src/models/reports/helpers/index.js

```diff
@@ -11,6 +44,53 @@ const whereInArray = (queryBuilder, column, parameter) => {
   }
 };
 
+/**
+ * Gets a report with subtotals.
+ *
+ * @param   {Array}          report            - The report data.
+ * @param   {Array}          subtotals         - The report subtotal data.
+ * @param   {string}         propertyToGroupBy - The property to group the report data by.
+ * @returns {Promise<Array>}                   An array of report Promise objects with subtotals added.
+ */
+const getReportWithSubtotals = async (report, subtotals, propertyToGroupBy) => {
+  // Group the report data by the specified property
+  const groupedReport = groupByProperty(report, propertyToGroupBy);
+
+  // Use reduce to fold in subtotals for each group
+  return _.reduce(
+    groupedReport,
+    (acc, report) =>
+      getNewReportGroup({
+        acc,
+        report,
+        subtotals,
+        propertyToGroupBy,
+      }),
+    // initial value - empty array to hold each new report group with subtotals as they accumulate
+    []
+  );
+};
+
+// helper utilities for getting a Report grouped by a property, with subtotals for each group
```

Comment:
add proper doc block please, also for getProjectName and getReportGroupSubtotals

Reply:
fixed